src/vm/fieldmarshaler.h

```diff
@@ -343,6 +345,18 @@ class FieldMarshaler
         m_nft = nft;
     }
 
+    NativeFieldCategory GetNativeFieldCategory() const
+    {
+        LIMITED_METHOD_CONTRACT;
+        return m_nfc;
+    }
+
+    void SetNativeFieldCategory(NativeFieldCategory nfc)
+    {
+        LIMITED_METHOD_CONTRACT;
+        m_nfc = nfc;
```

Comment:
Any invariant we can assert here? e.g. initial state of `m_nfc`?

Reply:
Since `m_nfc` is set as part of the `INITFIELDMARSHALER` macro (the only place field marshalers are created), I think it's fine the way it is right now. It's matching the m_nft field's initialization pattern.

Reply:
I disagree. This is a public function - that really shouldn't be - that should only be called once. I would assert that its initial value is set so we don't call this twice. Unless for some reason this isn't being initialized during construction.

Reply:
Added an assert.